src/core/pratt.ml

```diff
@@ -0,0 +1,113 @@
+(** This module defines functions needed by the Pratt parser of the Pratter
+    library. *)
+
+open Syntax
+
+module Pratt : sig
+  val parse : Sig_state.t -> Env.t -> p_term -> p_term
+end = struct
+
+  open Lplib
+  open Extra
+  open Pos
+
+  (** [find_qid ss env qid] fetches the qualified identifier associated to
+      [qid] if [qid] is the identifier of an infix operator defined in
+      signature state [ss]. If [qid] is defined in environment [env], then it
+      designates a variable, and is thus returned as-is. *)
+  (* REVIEW: this function relies on a lot of invariants and performs many
+     [Map.find_opt]s. Things may be optimised by modifying the data structure
+     of the signature state, notably making the qualified identfier to which
+     operators are linked available in the [unops] field of the signature
+     state. *)
+  let find_qid : Sig_state.t -> Env.t -> qident -> qident =
+    fun ss env ({elt=(mp, s); _} as qid) ->
+    try
+      if mp <> [] then raise Not_found;
+      ignore (Env.find s env);
+      qid
+    with Not_found ->
+    match StrMap.find_opt s ss.Sig_state.unops with
+    | Some sym -> (
+        match Terms.SymMap.find_opt sym ss.Sig_state.pp_hints with
+        | Some (Prefix(_, _, qid)) -> qid
+        | _ -> assert false )
+    | None -> (
+        match StrMap.find_opt s ss.Sig_state.binops with
+        | Some sym -> (
+            match Terms.SymMap.find_opt sym ss.Sig_state.pp_hints with
+            | Some (Infix (_, _, _, qid)) -> qid
+            | _ -> assert false )
+        | None -> qid )
+
+
+  module Pratt_terms : Pratter.SUPPORT
```

Comment:
@gabrielhdt Could you please add an URL to the definition of SUPPORT?

Reply:
Currently, this does not match https://github.com/gabrielhdt/pratter/ ...
Perhaps, while this is still under development, it would easier and faster to include pratter.ml, and later remove it and use the opam version instead?

Reply:
I've published the version used now, so the repository is up to date. I'll add the file anyway.

Reply:
Thanks @gabrielhdt . It would be still very useful to add an URL here anyway, or at the top of the file, using `@see <...>`.